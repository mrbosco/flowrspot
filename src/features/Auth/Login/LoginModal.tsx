import Modal from '../../../components/Modal';

const LoginModal = () => {
  return <Modal name="login">Access Account</Modal>;
};

export default LoginModal;
