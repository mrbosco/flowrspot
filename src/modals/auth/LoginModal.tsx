import Modal from '../../components/Modal';
import LoginForm from '../../features/Auth/LoginForm';

const LoginModal = () => {
  return (
    <Modal name="login">
      <LoginForm />
    </Modal>
  );
};

export default LoginModal;
