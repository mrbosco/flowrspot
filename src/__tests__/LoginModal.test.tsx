import { render } from '@testing-library/react';
import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
import LoginModal from '../modals/auth/LoginModal';

jest.mock('../stores/useModalStore', () => ({
  __esModule: true,
  default: jest.fn(() => ({
    modals: { login: true },
    openModal: jest.fn(),
    closeModal: jest.fn(),
  })),
}));

describe('LoginModal', () => {
  it('renders correctly and matches snapshot', () => {
    const queryClient = new QueryClient();

    const { asFragment } = render(
      <QueryClientProvider client={queryClient}>
        <LoginModal />
      </QueryClientProvider>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
