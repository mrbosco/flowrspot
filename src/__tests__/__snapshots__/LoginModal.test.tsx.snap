// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginModal renders correctly and matches snapshot 1`] = `<DocumentFragment />`;
